### ISSUE TITLE
Error with EVP_sha1 / OpenSSL

### ISSUE THREAD

**Issue by henriquechehad** [adobe-apiplatform/api-gateway-aws#19]
I'm trying to use the project but it's returning the error below:

```
Error → /usr/local/share/lua/5.1/api-gateway/resty/hmac.lua @ 63
suite ./test/specs/kinesis_spec.lua
/usr/local/share/lua/5.1/api-gateway/resty/hmac.lua:63: /usr/bin/luajit: undefined symbol: EVP_sha1
```

In my code i'm just importing, using:
```
local KinesisService = require('api-gateway.aws.kinesis.KinesisService')
```

I have the openssl, libcrypto, libssl-dev packages installed

**Comment by ddragosd** [comment]
Hi, what OpenSSL version do  you have installed ? It's either a missing header file or a dynamic
Linking issue

**Comment by henriquechehad** [comment]
Hi @ddragosd 
```# openssl version
OpenSSL 1.0.1f 6 Jan 2014```

**Comment by ddragosd** [comment]
Do you have a file named evp.h on the system ? Also, how are you building Openresty ?

**Comment by henriquechehad** [comment]
yes,

/usr/include/openssl/evp.h

I rebuilded openresty now from source, but same problem

**Comment by ddragosd** [comment]
Thanks, can you check if that evp.h file has the EVP_sha1 method defined ?
(I'm trying to help you from my phone only as I'm on the go today so I may
ask more q). If yes, mind pasting the signature if the method here ?

**Comment by henriquechehad** [comment]
in file exists this content with EVP_sha1:

```
#ifndef OPENSSL_NO_SHA
const EVP_MD *EVP_sha(void);
const EVP_MD *EVP_sha1(void);
const EVP_MD *EVP_dss(void);
const EVP_MD *EVP_dss1(void);
const EVP_MD *EVP_ecdsa(void);
#endif
```

Is missing something?

**Comment by henriquechehad** [comment]
I tested in ubuntu and debian docker containers, same issue in both

**Comment by ddragosd** [comment]
The signature is correct and is the same as the one used here :
https://github.com/adobe-apiplatform/api-gateway-hmac/blob/master/src/lua/api-gateway/resty/hmac.lua

So right now it seems to be a dynamic linking issue in the OS with OpenSSL.

**Comment by ddragosd** [comment]
Is there a sample you can share with me ? I'd be happy to take a look over the weekend or early next week. It shouldn't be something too complicated. Another option, in the meanwhile, is to try with this docker image : https://hub.docker.com/r/adobeapiplatform/apigateway/ , built from https://github.com/adobe-apiplatform/apigateway .

**Comment by henriquechehad** [comment]
@ddragosd thanks for helping. the issue was because i was trying to test with busted, but i realized now, the project runs only inside nginx

**Comment by henriquechehad** [comment]
@ddragosd is there some reason for this project is tied in nginx? is not viable to create an independent version?

**Comment by ddragosd** [comment]
@enriquechehad no special reason other than being impl initially to work with NGINX / Openresty. Are you saying you'd like to use this module w/o NGINX ?

**Comment by henriquechehad** [comment]
@ddragosd yes, api-gateway-aws is the most complete Lua project for AWS that I found. Would be good to have the possibility to use the project without NGINX.

**Comment by ddragosd** [comment]
@henriquechehad, on a second thought it doesn't seem to be that straight forward to decouple it from NGINX as this module depends on multiple openresty features. At the same time I have to admit you made me curious if there are cases when such a lua module would be useful, comparing with the other [aws sdks](https://github.com/aws?utf8=%E2%9C%93&q=aws-sdk&type=&language=).
